ios/RCCNavigationController.m

```diff
@@ -211,6 +211,8 @@ - (void)performAction:(NSString*)performAction actionParams:(NSDictionary*)actio
     
     RCCViewController *viewController = [[RCCViewController alloc] initWithComponent:component passProps:passProps navigatorStyle:navigatorStyle globalProps:nil bridge:bridge];
     
+    viewController.navigationItem.hidesBackButton = YES;
```

Comment:
I don't think it's about setting the correct JS properties for the next controller - the back button flickers regardless of the back button props coming through the bridge. I was under the assumption that it was normal to have to specify hiding the back button when performing any routing.  

I do agree with the last statement, however I don't have the ObjC knowledge / the time to diagnose issues with the routing flow - I just know this works.